source/rust_verify/src/rust_to_vir_base.rs

```diff
@@ -1069,3 +1069,20 @@ pub(crate) fn check_generics_bounds_fun<'tcx>(
             .collect(),
     ))
 }
+
+pub(crate) fn auto_deref_supported_for_ty<'tcx>(
+    tcx: TyCtxt<'tcx>,
+    ty: &rustc_middle::ty::Ty<'tcx>,
+) -> bool {
+    match ty.kind() {
+        TyKind::Adt(AdtDef(adt_def_data), _args) => {
+            let did = adt_def_data.did;
+            if Some(did) == tcx.lang_items().owned_box() {
+                return true;
+            }
+            let rust_item = verus_items::get_rust_item(tcx, did);
+            return matches!(rust_item, Some(RustItem::Box | RustItem::Rc | RustItem::Arc));
```

Comment:
I think the check with `owned_box` is redundant (the `matches!` `RustItem::Box` should have the same effect).

```suggestion
            let rust_item = verus_items::get_rust_item(tcx, did);
            return matches!(rust_item, Some(RustItem::Box | RustItem::Rc | RustItem::Arc));
```